src/v/cloud_roles/signature.cc

```diff
@@ -60,20 +60,20 @@ time_source::time_source()
 time_source::time_source(timestamp instant)
   : time_source([instant]() { return instant; }, 0) {}
 
-ss::sstring time_source::format(const char* fmt) const {
-    std::array<char, formatted_datetime_len> out_str{};
-    auto point = _gettime_fn();
-    std::time_t time = std::chrono::system_clock::to_time_t(point);
-    std::tm* gm = std::gmtime(&time);
-    auto ret = std::strftime(out_str.data(), out_str.size(), fmt, gm);
-    vassert(ret > 0, "Invalid date format string");
-    return ss::sstring(out_str.data(), ret);
+ss::sstring time_source::format_date() const {
+    return format<formatted_date_len>("%Y%m%d");
 }
 
-ss::sstring time_source::format_date() const { return format("%Y%m%d"); }
-
 ss::sstring time_source::format_datetime() const {
-    return format("%Y%m%dT%H%M%SZ");
+    return format<formatted_datetime_len>("%Y%m%dT%H%M%SZ");
+}
+
+ss::sstring time_source::format_http_datetime() const {
+    // Non english locales break this, but the equivalent of
```

Comment:
given that fmt 8.1.1 supports locale independence and chrono formatting, maybe it would be good to take this opportunity to switch to it, also to remove this out of band dependency on std::setlocale(LC_ALL, "C")

Reply:
Thanks for the review and the great suggestion. I switched the time formatting to use fmt in: https://github.com/redpanda-data/redpanda/pull/8055/commits/ec594d8a360aecd2bb5ff4d14b3bf60cfcdbcfc1.